LanguageFeatures/Patterns/matching_record_A01_t02.dart

```diff
@@ -63,9 +63,9 @@ const c4 = C(4, logger);
 
 String test1(Object o) {
   switch (o) {
-    case (C c1, C c2):
+    case (c1, c2):
```

Comment:
The run-time type of the matched object here is `(Null, int)` and the required type of the record pattern is `(Object?, Object?)`: the required type of a record pattern [_only_ tests the shape](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=the%20matched%20value%20type.-,The%20required%20type,-of%20p%20is%20a).

The required types of the field patterns is not mentioned explicitly, but we have no reason to assume that it is different from `Object?`. The matching process will check for `theConstant == theMatchedValue`, and there is no need to perform a type check first (actually, the type check could be wrong, because `theConstant` could claim that it is `==` to absolutely anything it wants, and then we shouldn't reject `theMatchedValue` because of its type).

This means that we still won't fail the match based on the required type as stated in the description.

I think the test will have to rely on different shapes, and on non-records, that is, stuff like `test1(true)`, `test1(())`, `test1((notNameAnything: 5,)`.

At the same time, this sounds like a kind of test which is already covered. Do you think this is just a duplicate of some existing tests? I know it's not very convenient to search through all tests, but perhaps a quick grep for 'shape' could reveal something?